test/test_ion_text.cpp

```diff
@@ -720,3 +720,426 @@ TEST(IonTextInt, BinaryLiterals) {
+
+    hREADER  reader;
+    ION_TYPE type;
+
+    ION_ASSERT_OK(ion_test_new_text_reader(ion_text, &reader));
+
+    int64_t offset = 0;
+    int32_t line = 0;
+    int32_t col_offset = 0;
+
+    // a::1
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_INT, type);
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 2);
+    ASSERT_EQ(line, 1);
+    ASSERT_EQ(col_offset, 2);
+
+    // b::2
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_INT, type);
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 7);
+    ASSERT_EQ(line, 1);
+    ASSERT_EQ(col_offset, 7);
+
+    // c::3
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_INT, type);
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 12);
+    ASSERT_EQ(line, 1);
+    ASSERT_EQ(col_offset, 12);
+
+}
+
+/** tests ion_reader_get_value_position for list elements appearing on separate lines. */
+TEST(IonTextPosition, PositionOfListElements) {
+    // this is `[1,2,3] 42` split across multiple lines with values prefixed with various tabs and spaces.
+    const char *ion_text = "\n[\n\t1,\n\t 2,\n\t  3\n]\n 42";
+
+    hREADER  reader;
+    ION_TYPE type;
+
+    ION_ASSERT_OK(ion_test_new_text_reader(ion_text, &reader));
+
+    int64_t offset = 0;
+    int32_t line = 0;
+    int32_t col_offset = 0;
+
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_LIST, type);
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 1);
+    ASSERT_EQ(line, 2);
+    ASSERT_EQ(col_offset, 0);
+
+    ION_ASSERT_OK(ion_reader_step_in(reader));
+
+    // value position should remain unchanged after step in
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 1);
+    ASSERT_EQ(line, 2);
+    ASSERT_EQ(col_offset, 0);
+
+    // first list element
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_INT, type);
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 4);
+    ASSERT_EQ(line, 3);
+    ASSERT_EQ(col_offset, 1);
+
+    // second list element
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_INT, type);
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 9);
+    ASSERT_EQ(line, 4);
+    ASSERT_EQ(col_offset, 2);
+
+    // third list element
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_INT, type);
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 15);
+    ASSERT_EQ(line, 5);
+    ASSERT_EQ(col_offset, 3);
+
+    // finally, step out
+    ION_ASSERT_OK(ion_reader_step_out(reader));
```

Comment:
What happens if you do another next() before stepping out to find the end of the container? Would be good to add verification here. (This is common when looping through all of a container's elements.)

Reply:
happens to be positioned on the closing `]`, `)` or `}`, which is convenient.  added tests and documentation.

Reply:
also added a similar call to test the position after `ion_reader_next()` encounters a legit EOF.